### PR TITLE
fix: update createmeta API

### PR DESCRIPTION
**Pull request by afdesk**
# Description

Please describe _what does this Pull Request fix or add?_.

Information that is useful here:
* **The What**: What is your change doing?
* **The Why**: Why is your change useful/needed? What is your use case? (this helps us to understand the real world better)
* **Type of change**: Things like Bugfix, New feature, Code quality improvements, Dependency upgrade, Documentation, ...
* **Breaking change**: Yes or no? Backward compatible?
* **Related to an issue**: Does this fix or close an issue? Or is related in any kind?
* **Jira Version + Type**: Which Jira version and type (on-premise / cloud) you have used?

## Example:

Let us know how users can use or test this functionality.

```go
// Example code

```

# Checklist

* [ ] Unit or Integration tests added
  * [ ] Good Path
  * [ ] Error Path
* [ ] Commits follow conventions described here:
  * [ ] [Conventional Commits 1.0.0](https://conventionalcommits.org/en/v1.0.0-beta.4/#summary)
  * [ ] [The seven rules of a great Git commit message](https://chris.beams.io/posts/git-commit/#seven-rules)
* [ ] Commits are squashed such that
  * [ ] There is 1 commit per isolated change
* [ ] I've not made extraneous commits/changes that are unrelated to my change.

### CODE REVIEW

**Review by roeelivaqua — APPROVED** [review]

File: metaissue.go [hunk 1]
```diff
@@ -148,19 +213,21 @@ func (p *MetaProject) GetIssueTypeWithName(name string) *MetaIssueType {
...
+//										"custom": "com.pyxis.greenhopper.jira:gh-epic-link",
+//										"customId": 10806
+//									},
+//									"name": "Epic Link",
+//									"hasDefaultValue": false,
+//									"operations": [
+//										"set"
+//									]
+//								}
+//
```
Comment (simar7) [inline metaissue.go]: Has something changed here? If not we should cleanup.

File: metaissue.go [hunk 2]
```diff
@@ -48,6 +48,18 @@ type MetaIssueType struct {
 	Fields      tcontainer.MarshalMap `json:"fields,omitempty"`
 }
 
+type MetaIssueTypeDetails struct {
+	IsLast     bool                    `json:"isLast,omitempty" yaml:"isLast,omitempty"`
+	MaxResults int                     `json:"maxResults,omitempty" yaml:"maxResults,omitempty"`
+	StartAt    int                     `json:"startAt,omitempty" yaml:"startAt,omitempty"`
+	Total      int                     `json:"total,omitempty" yaml:"total,omitempty"`
+	Values     []tcontainer.MarshalMap `json:"values,omitempty"`
```
Comment (simar7) [inline metaissue.go:56]: missing yaml tag? or does it not need one?

File: metaissue.go [hunk 3]
```diff
@@ -48,6 +48,18 @@ type MetaIssueType struct {
...
+}
+
+type JiraServerInfo struct {
+	VersionNumbers []int `json:"versionNumbers"`
```
Comment (simar7) [inline metaissue.go:60]: ditto for yaml tags

File: metaissue.go [hunk 4]
```diff
@@ -58,10 +70,65 @@ func (s *IssueService) GetCreateMeta(projectkeys string) (*CreateMetaInfo, *Resp
...
+	if err != nil {
+		return 0
+	}
+	version := new(JiraServerInfo)
+
+	_, err = s.client.Do(req, version)
+	if err != nil {
+		return 0
+	}
+	return version.VersionNumbers[0]
```
Comment (simar7) [inline metaissue.go:129]: What else is in this slice? Could we document it here maybe?

**Review by simar7 — APPROVED** [review]
left some comments, they are mostly nits so overall lgtm.

File: metaissue.go (hunk 2 above)
Comment (afdesk) [reply metaissue.go:56]: nice catch! we don't need yaml here for Jira API structs.

File: metaissue.go (hunk 3 above)
Comment (afdesk) [reply metaissue.go:60]: yaml tags were mistake

File: metaissue.go (hunk 1 above)
Comment (afdesk) [reply metaissue.go]: the linter corrects it. fixed

File: metaissue.go (hunk 4 above)
Comment (afdesk) [reply metaissue.go:129]: add comments.

**Comment by afdesk** [comment]
> left some comments, they are mostly nits so overall lgtm.

@simar7 thanks for the review. I added some comments and changes.

**Review by simar7 — APPROVED** [review]

**Pull request merged**